lib/spack/spack/patch.py

```diff
@@ -372,14 +499,29 @@ def update_package(self, pkg_fullname):
             p2p = self.index.setdefault(sha256, {})
             p2p.update(package_to_patch)
 
-    def update(self, other):
-        """Update this cache with the contents of another."""
+    def update(self, other: "PatchCache") -> None:
+        """Update this cache with the contents of another.
+
+        Args:
+            other: another patch cache to merge
+        """
         for sha256, package_to_patch in other.index.items():
             p2p = self.index.setdefault(sha256, {})
             p2p.update(package_to_patch)
 
     @staticmethod
-    def _index_patches(pkg_class, repository):
+    def _index_patches(
+        pkg_class: "spack.package_base.PackageBase", repository: "spack.repo.RepoPath"
```

Comment:
I think we need to import `Type` from `typing` and:
```suggestion
        pkg_class: Type["spack.package_base.PackageBase"], repository: "spack.repo.RepoPath"
```